pycbc/inference/models/gated_gaussian_noise.py

```diff
@@ -154,6 +141,128 @@ def _loglr(self):
         """
         return self.loglikelihood - self.lognl
 
+    def whiten(self, data, whiten, inplace=False):
+        """Whitens the given data.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary of detector names -> FrequencySeries.
+        whiten : {0, 1, 2}
+            Integer indicating what level of whitening to apply. Levels are:
+            0: no whitening; 1: whiten; 2: overwhiten.
+        inplace : bool, optional
+            If True, modify the data in place. Otherwise, a copy will be
+            created for whitening.
+
+
+        Returns
+        -------
+        dict :
+            Dictionary of FrequencySeries after the requested whitening has
+            been applied.
+        """
+        if not inplace:
+            data = {det: d.copy() for det, d in data.items()}
+        if whiten:
+            for det, dtilde in data.items():
+                invpsd = self._invpsds[det]
+                if whiten == 1:
+                    dtilde *= invpsd**0.5
+                elif whiten == 2:
+                    dtilde *= invpsd
+                else:
+                    raise ValueError("whiten must be either 0, 1, or 2")
+        return data
+
+    def get_waveforms(self):
+        """The waveforms generated using the current parameters.
+
+        If the waveforms haven't been generated yet, they will be generated
+        and cached.
+
+        Returns
+        -------
+        dict :
+            Dictionary of detector names -> FrequencySeries.
+        """
+        if self._current_wfs is None:
+            params = self.current_params
+            self._current_wfs = self.waveform_generator.generate(**params)
+        return self._current_wfs
+
+
+    @abstractmethod
+    def get_gated_waveforms(self):
+        """Generates and gates waveforms using the current parameters.
+
+        Returns
+        -------
+        dict :
+            Dictionary of detector names -> FrequencySeries.
+        """
+        pass
+
+    def get_residuals(self):
+        """Generates the residuals ``d-h`` using the current parameters.
+
+        Returns
+        -------
+        dict :
+            Dictionary of detector names -> FrequencySeries.
+        """
+        wfs = self.get_waveforms()
+        out = {}
+        for det, h in wfs.items():
+            invpsd = self._invpsds[det]
```

Comment:
CC is reporting this as unused, and it certainly looks like it is to me.

Reply:
Fixed.